tools/distpartitioning/convert_partition.py

```diff
@@ -65,44 +78,46 @@ def _get_unique_invidx(srcids, dstids, nids):
         a list of integers. These are inverse indices, which will be indices
         from the unique elements list specifying the elements from the
         input array, dstids
-
     """
-    mask = np.in1d(srcids, nids, invert=True, kind="table")
+    assert len(srcids) == len(
+        dstids
+    ), f"Please provide the correct input parameters"
+    assert len(srcids) != 0, f"Please provide a non-empty edge-list."
+
+    if np.__version__ < "1.24.0":
+        logging.warning(
+            f"Numpy version, {np.__version__}, is lower than expected."
+            f"Falling back to numpy's native function unique."
+            f"This functions memory overhead will limit size of the "
+            f"partitioned graph objects processed by each node in the cluster."
+        )
+        uniques, idxes, inv_idxes = np.unique(
+            np.concatenate(srcids, dstids, nids),
+            return_index=True,
+            return_inverse=True,
+        )
+        src_len = len(srcids)
+        dst_len = len(dstids)
+        return (
+            uniques,
+            idxes,
+            inv_idxes[:src_len],
+            inv_idxes[src_len : (src_len + dst_len)],
+        )
+
+    # find uniqes which appear only in the srcids list
+    mask = np.isin(srcids, nids, invert=True, kind="table")
     srcids_only = srcids[mask]
     srcids_idxes = np.where(mask == 1)[0]
 
-    # sort the array
+    # sort
     uniques, unique_srcids_idx = np.unique(srcids_only, return_index=True)
     idxes = srcids_idxes[unique_srcids_idx]
 
-    # store this for testing
-    gold_uniques = np.concatenate([uniques, nids])
-
-    # compute the idxes return parameter
-    # Remaining elements in the srcids array
-    srcids_int_nids, comm1, _ = np.intersect1d(
-        srcids, nids, return_indices=True
-    )
-    idxes = np.concatenate([idxes, comm1])
-    uniques = np.concatenate([uniques, srcids_int_nids])
-
-    # Remaining elements in the dstids array
-    dstids_int_nids, comm1, _ = np.intersect1d(
-        dstids, nids, return_indices=True
-    )
-    idxes = np.concatenate([idxes, comm1])
-    uniques = np.concatenate([uniques, dstids_int_nids])
-    assert len(dstids_int_nids) <= len(nids), (
-        f"node-ids array does not meet the requirements "
-        f"It is not a super-set. "
-        f"len(dstids_int_nids) = {len(dstids_int_nids)}, len(nids) = {len(nids)}"
-    )
-
-    # validation check here.
-    assert np.all(gold_uniques.sort() == uniques.sort()), (
-        f"Our assumption about gold_unique is NOT right. "
-        f"len(gold_uniques) == {len(gold_uniques)} "
-        f"len(uniques) == {len(uniques)}"
+    # build uniques and idxes, first and second return parameters
```

Comment:
Did you change the implementation?

Reply:
In a way yes. This newer implementation is optimized for wall-clock time. In the earlier version, which was used for benchmarking purposes on felarge and webgraphs, we used to compute the list of uniques in a step by step manner - and validating them as well. Now I removed the validation part, since the benchmarking results were highly reduced when compared to the numpy's unique implementation.